node/src/Pos/Block/Logic/Header.hs

```diff
@@ -328,35 +327,84 @@ getHeadersOlderExp upto = do
                 | otherwise = selGo es ii $ succ skipped
         in selGo elems ixs 0
 
--- CSL-396 don't load all the blocks into memory at once
--- | Given @from@ and @to@ headers where @from@ is older (not strict)
--- than @to@, and valid chain in between can be found, headers in
--- range @[from..to]@ will be found.
-getHeadersFromToIncl
-    :: forall ssc m .
-       (HasConfiguration, MonadDBRead m, SscHelpersClass ssc)
-    => HeaderHash -> HeaderHash -> m (Maybe (OldestFirst NE HeaderHash))
-getHeadersFromToIncl older newer = runMaybeT . fmap OldestFirst $ do
+-- | Given optional @depthLimit@, @from@ and @to@ headers where @from@
+-- is older (not strict) than @to@, and valid chain in between can be
+-- found, headers in range @[from..to]@ will be found. If difference
+-- between headers is more than @depthLimit@, error will be thrown.
```

Comment:
Imo, this interface is strange. I find it more intuitive to limit number of headers.
Also there is a problem: eventually you indeed want to limit number of headers, but since you limit difference in difficulty it can happen that there are more headers, because there is a genesis header in between.

Reply:
@gromakovsky nice catch. I'll fix that.

Reply:
`difference between headers` is really obscure :/
Do you mean difference between their `ChainDiffuclties` or just number of headers?
I can figure out by looking at the implementation, but the documentation should be enough to understand the meaning of the first argument.